core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java

```diff
@@ -239,7 +239,9 @@ private static MachineConfigImpl asMachineConfig(Document document) {
                                                                   .setType(document.getString("type"));
         final Document sourceDocument = document.get("source", Document.class);
         if (sourceDocument != null) {
-            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type"), sourceDocument.getString("location")));
+            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type"))
+                                      .setContent(sourceDocument.getString("content"))
+                                      .setLocation(sourceDocument.getString("location")));
```

Comment:
Following your usage of constants for fields consider moving **MACHINE_SOURCE...** constants to this class and usage them in this class and SnapshotImplCodec